common/src/main/java/com/wynntils/core/functions/arguments/FunctionArguments.java

```diff
@@ -154,5 +188,23 @@ public Double getDoubleValue() {
         public String getStringValue() {
             return (String) this.getValue();
         }
+
+        public <U> ListArgument<U> asList() {
+            return (ListArgument<U>) this;
+        }
+    }
+
+    public static class ListArgument<T> extends Argument<T> {
+        public ListArgument(String name, Class<T> type) {
+            super(name, type, null);
+        }
+
+        public void setValues(List<Object> values) {
```

Comment:
```suggestion
        public void setValues(List<T> values) {
```
?

Reply:
setValue itself uses object too, i followed that

Reply:
Or maybe not, I see that you don't know formally the type. Nevermind me, I'm just throwing bad suggestions around me tonight. :)